operator/pkg/proxy/translation.go

```diff
@@ -53,6 +53,35 @@ func InjectProxyEnvVars(translator values.Translator, proxyEnv map[string]string
 		}
 
 		proxyVals, _ := getProxyConfigHelmValues(obj, proxyEnv) // ignore errors for now
-		return chartutil.CoalesceTables(vals, proxyVals), nil
+
+		mergedVals := chartutil.CoalesceTables(vals, proxyVals)
+		mergedVals = delValueFromIfValueExists(mergedVals)
+
+		return mergedVals, nil
 	})
 }
+
+// delValueFromIfValueExists deletes the valueFrom key from customize.envVars entries
```

Comment:
Would it be possible to prefer our proxy configurations instead of the global proxy if the CR defines them?

Reply:
Currently if both env and CR set the proxy as value. The config from CR is used.

I don't know what happens if the CR sets a valueFrom. But that should be a good edge case to add to the tests. Will send an update once I tried that.

Reply:
Nice, than it is already the desired behaviour. 
Does the openshift global proxy overwrite the env variables on the deployment/pods as well? I am worried that the Helm chart is applied correctly but OpenShift overwrites our previously applied config happily :)

Reply:
Added a log in case of value conflicts.